packages/cohort_sdk/src/cohort.rs

```diff
@@ -360,49 +405,40 @@ impl Cohort {
             });
         }
 
-        result
+        final_result
     }
 
-    async fn send_to_talos_attempt<S>(
-        &self,
-        request: model::CertificationRequest,
-        state_provider: &S,
-        previous_conflict: Option<u64>,
-    ) -> CertificationAttemptOutcome
+    async fn send_to_talos_attempt<F, Fut>(&self, generate_certifier_candidate_callback: &F, previous_conflict: Option<u64>) -> CertificationAttemptOutcome
     where
-        S: ItemStateProvider,
+        F: Fn() -> Fut,
+        Fut: Future<Output = Result<CertificationCandidateCallbackResponse, String>>,
     {
-        let timeout = if request.timeout_ms > 0 {
-            Duration::from_millis(request.timeout_ms)
-        } else {
-            Duration::from_millis(self.config.snapshot_wait_timeout_ms as u64)
-        };
+        let timeout = Duration::from_millis(self.config.snapshot_wait_timeout_ms as u64);
 
-        let local_state: CapturedState = match self.await_for_snapshot(state_provider, previous_conflict, timeout).await {
+        let request = match self.await_for_snapshot(generate_certifier_candidate_callback, previous_conflict, timeout).await {
             Err(SnapshotPollErrorType::FetchError { reason }) => return CertificationAttemptOutcome::DataError { reason },
             Err(SnapshotPollErrorType::Timeout { waited }) => {
                 return CertificationAttemptOutcome::SnapshotTimeout {
                     waited,
                     conflict: previous_conflict.unwrap(),
                 }
             }
-            Ok(local_state) => local_state,
+            Ok(CertificationCandidateCallbackResponse::Cancelled(reason)) => {
+                return CertificationAttemptOutcome::ClientAborted { reason };
```

Comment:
There are too many changes in this class, I might be getting confused here. Aren't we doing something like this? From Cancelled to ClientAborted and then again to Cancelled?  If yes, maybe we should use single naming approach, either all Cancelled or all ClientAborted :)

Reply:
good catch, that should be cancelled. I missed that `ClientAbort` when I changed the rest. Will update it